typing/typecore.ml

```diff
@@ -4377,6 +4376,29 @@ and type_expect_
       | _ ->
           raise (Error (loc, env, Invalid_extension_constructor_payload))
       end
+  | Pexp_extension ({ txt = ("ocaml.atomic.loc"
+                             |"atomic.loc"); _ },
+                    payload) ->
+      begin match payload with
+      | PStr [ { pstr_desc =
+                  Pstr_eval (
+                    { pexp_desc = Pexp_field (srecord, lid); _ } as sexp, _
+                  )
+               } ] ->
+          let record, label, ty_arg =
+            solve_Pexp_field ~label_usage:Env.Mutation env sexp srecord lid
```

Comment:
When no change was made, could you please not mark as resolved without an explanation? I’m sure you have reasons to consider this point minor, but then I have to figure out whether my comment was made irrelevant by later changes, or whether you just disagree and decided to silently dismiss my suggestion.

Reply:
I see, I had missed this. With the `Env.Projection` there and the `Env.Mutation` above, am I correct in understanding that there should never be a warning “record field `foo` is never used/mutated” if there is at least one `[%atomic.loc x.foo]` in the program?

Reply:
Yes, this is the intent: mark those atomic-loc expressions as performing both read- and write-access to the field. (Because they can perform either, most likely both, and we cannot easily track which.)